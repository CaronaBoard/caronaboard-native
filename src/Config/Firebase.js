import Firebase from 'firebase'

const config = {
  apiKey: 'AIzaSyDfwHLwWKTqduazsf4kjbstJEA2E1sCeoI',
  authDomain: 'caronaboard-61f75.firebaseapp.com',
  databaseURL: 'https://caronaboard-61f75.firebaseio.com',
  storageBucket: 'caronaboard-61f75.appspot.com',
  messagingSenderId: '617045704123'
}

const setup = () => {
  console.log('firebase setup ' + Firebase)
  return Firebase.initializeApp(config)
}

export default { setup }
