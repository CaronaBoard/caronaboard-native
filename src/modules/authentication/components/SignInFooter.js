import React, { Component } from 'react'
import { View } from 'react-native'
import PropTypes from 'prop-types'
import { RkButton, RkText } from 'react-native-ui-kitten'
import Styles from './styles/SignInFooterStyles'
import { screens } from '../../../configuration/navigation/Screens'

export class SignInFooter extends Component {
  static propTypes = {
    email: PropTypes.string,
    navigator: PropTypes.object.isRequired
  }

  navigateTo = screen => () => this.props.navigator.push({screen})

  render () {
    return (
      <View>
        <View style={Styles.textRow}>
          <RkButton
            key='forgot-password-button'
            rkType='clear'
            onPress={this.navigateTo(screens.forgotPassword.id)}
          >
            <RkText rkType='header6'>Forgot your password?</RkText>
          </RkButton>
        </View>
        <View style={Styles.textRow}>
          <RkText rkType='primary3'>Donâ€™t have an account?</RkText>
          <RkButton
            key='signup-button'
            rkType='clear'
            onPress={this.navigateTo(screens.signUp.id)}
          >
            <RkText rkType='header6'> Sign up now</RkText>
          </RkButton>
        </View>
      </View>
    )
  }
}
