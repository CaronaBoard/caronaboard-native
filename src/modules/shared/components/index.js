export { Button } from './Button'
export { TextInput } from './TextInput'
