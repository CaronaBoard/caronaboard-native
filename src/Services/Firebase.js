import Firebase from 'firebase'
import _ from 'lodash'

const config = {
  apiKey: 'AIzaSyDfwHLwWKTqduazsf4kjbstJEA2E1sCeoI',
  authDomain: 'caronaboard-61f75.firebaseapp.com',
  databaseURL: 'https://caronaboard-61f75.firebaseio.com',
  storageBucket: 'caronaboard-61f75.appspot.com',
  messagingSenderId: '617045704123'
}

let database = (() => {
  Firebase.initializeApp(config)
  return Firebase.database().ref()
})()

const toArrayOfRides = (firebaseResponse) => {
  return _.flatMap(firebaseResponse, Object.values)
}

export const getAllRides = () => {
  return new Promise(resolve => {
    database.child('rides').once('value').then(
      (snapshot) => resolve(toArrayOfRides(snapshot.val()))
    )
  })
}
