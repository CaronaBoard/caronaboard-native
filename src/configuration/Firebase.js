import Firebase from 'firebase'

const config = {
  apiKey: 'AIzaSyDfwHLwWKTqduazsf4kjbstJEA2E1sCeoI',
  authDomain: 'caronaboard-61f75.firebaseapp.com',
  databaseURL: 'https://caronaboard-61f75.firebaseio.com',
  storageBucket: 'caronaboard-61f75.appspot.com',
  messagingSenderId: '617045704123'
}

const initialize = () => {
  Firebase.initializeApp(config)
}

export default { initialize }
