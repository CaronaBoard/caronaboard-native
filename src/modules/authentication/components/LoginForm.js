import React from 'react'
import { View, Image, Keyboard } from 'react-native'
import Toast, { DURATION } from 'react-native-easy-toast'
import { RkText, RkAvoidKeyboard } from 'react-native-ui-kitten'
import { GradientButton, TextInput } from '../../shared/components'
import { styles } from '../containers/styles/LoginScreenStyles'

export class LoginForm extends React.Component {
  static defaultProps = {
    footer: () => {},
    toast: {
      showAlert: false,
      message: ''
    }
  }

  state = {
    email: '',
    password: '',
    loading: false
  }

  onButtonPress = () => {
    const { email, password } = this.state
    this.props.onButtonPress(email, password)
  }

  componentWillReceiveProps (nextProps) {
    const { showAlert, message } = nextProps.toast

    if (showAlert) {
      this.refs.toast.show(message, DURATION.LENGTH_LONG)
    }
  }

  render () {
    return (
      <RkAvoidKeyboard
        style={styles.screen}
        onStartShouldSetResponder={e => true}
        onResponderRelease={e => Keyboard.dismiss()}>

        <View style={styles.header}>
          <Image style={styles.image} source={require('../../../assets/images/caronaboard-logo-azul.png')} />
          <RkText rkType='logo h0'>Caronaboard</RkText>
          <RkText rkType='light h1'>Awesome Slogan</RkText>
        </View>

        <TextInput
          onChangeText={(email) => { this.setState({email}) }}
          key='email-input'
          rkType='rounded'
          placeholder='Username'
          autoCapitalize='none'
        />
        <TextInput
          onChangeText={(password) => { this.setState({password}) }}
          key='password-input'
          rkType='rounded'
          placeholder='Password'
          secureTextEntry
        />

        <View style={styles.centralized}>
          <GradientButton
            rkType='large'
            style={styles.loginButton}
            text={this.props.buttonText}
            onPress={this.onButtonPress}
          />
        </View>

        {this.props.footer(this.state.email)}
        <Toast position='top' ref='toast' />
      </RkAvoidKeyboard>
    )
  }
}
