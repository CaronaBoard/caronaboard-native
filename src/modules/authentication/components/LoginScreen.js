import React from 'react'
import { View, Image, Keyboard } from 'react-native'
import Toast, { DURATION } from 'react-native-easy-toast'
import { RkText, RkTextInput, RkAvoidKeyboard } from 'react-native-ui-kitten'
import { GradientButton } from '../../shared/components/index'
import { styles } from '../containers/styles/LoginScreenStyles'

export const INITIAL_STATE = {
  email: '',
  password: '',
  loading: false
}

export default class LoginV2 extends React.Component {
  constructor (props) {
    super(props)
    this.state = INITIAL_STATE
  }

  onButtonPress = () => {
    const { email, password } = this.state
    const { onButtonPress } = this.props
    try {
      onButtonPress(email, password)
    } catch (error) {
      console.log(error)
    }
  }

  componentWillReceiveProps (nextProps) {
    const { showAlert, message } = nextProps.alert

    if (showAlert) {
      this.refs.toast.show(message, DURATION.LENGTH_LONG)
    }
  }

  render () {
    return (
      <RkAvoidKeyboard
        style={styles.screen}
        onStartShouldSetResponder={e => true}
        onResponderRelease={e => Keyboard.dismiss()}>

        <View style={styles.header}>
          <Image style={styles.image} source={require('../../../assets/images/logo.png')} />
          <RkText rkType='logo h0'>Caronaboard</RkText>
          <RkText rkType='light h1'>Awesome Slogan</RkText>
        </View>

        <RkTextInput onChangeText={(email) => { this.setState({email}) }} rkType='rounded' placeholder='Username' autoCapitalize='none' />
        <RkTextInput onChangeText={(password) => { this.setState({password}) }} rkType='rounded' placeholder='Password' secureTextEntry />

        <View style={styles.centralized}>
          <GradientButton
            style={styles.loginButton}
            rkType='large'
            text='LOGIN'
            onPress={this.onButtonPress}
          />
        </View>

        {this.props.footer()}
        <Toast position='top' ref='toast' />
      </RkAvoidKeyboard>
    )
  }
}
