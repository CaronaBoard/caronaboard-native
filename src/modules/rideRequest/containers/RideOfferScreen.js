import React, { Component } from 'react'
import { View } from 'react-native'
import { RkText } from 'react-native-ui-kitten'

import { Button, TextInput } from '../../shared/components'
import { saveRideOffer } from '../../../services/firebase'
import styles from './styles/RideOfferScreenStyles'

export const INITIAL_STATE = {
  origin: '',
  destination: '',
  days: '',
  hours: ''
}

export class RideOfferScreen extends Component {
  constructor (props) {
    super(props)
    this.state = INITIAL_STATE
  }

  render () {
    return (
      <View style={styles.flexible}>
        <View style={styles.container}>
          <RkText style={styles.title}>
            Ride Offer
          </RkText>
          <View style={styles.inputTextsContainer}>
            <TextInput placeholder='ORIGIN' onChangeText={(origin) => this.setState({origin})} />
            <TextInput placeholder='DESTINATION' onChangeText={(destination) => this.setState({destination})} />
            <TextInput placeholder='DAYS' onChangeText={(days) => this.setState({days})} />
            <TextInput placeholder='HOUR' onChangeText={(hours) => this.setState({hours})} />
          </View>
          <Button text='Offer Ride' onPress={() => saveRideOffer(this.state)} />
        </View>
      </View>
    )
  }
}

export default RideOfferScreen
