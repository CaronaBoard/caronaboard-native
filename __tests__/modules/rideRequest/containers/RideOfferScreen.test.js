import React from 'react'
import { shallow } from 'enzyme'
import { RideOfferScreen, INITIAL_STATE } from '../../../../src/modules/rideRequest/containers/RideOfferScreen'
import { Button } from '../../../../src/modules/shared/components'

import * as Firebase from '../../../../src/services/Firebase'
jest.mock('../../../../src/services/Firebase')

describe('<RideOfferScreen />', () => {
  const wrapper = shallow(<RideOfferScreen />)

  it('Snapshot have a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Should call firebase service for create a ride offer', () => {
    // TODO: I Think that' not a good idea calling services directly, we should use action
    Firebase.saveRideOffer = jest.fn()
    wrapper.find(Button).simulate('press')
    expect(Firebase.saveRideOffer).toHaveBeenCalledWith(INITIAL_STATE)
  })
})
