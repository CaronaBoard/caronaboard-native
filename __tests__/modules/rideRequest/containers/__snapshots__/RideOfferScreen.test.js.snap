// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RideOfferScreen /> Snapshot have a snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "space-around",
        "paddingHorizontal": 50,
      }
    }
  >
    <RkText
      style={
        Object {
          "fontSize": 38,
          "textAlign": "center",
        }
      }
    >
      Ride Offer
    </RkText>
    <View
      style={
        Object {
          "minHeight": 370,
        }
      }
    >
      <TextInput
        label="ORIGIN"
        onChangeText={[Function]}
      />
      <TextInput
        label="DESTINATION"
        onChangeText={[Function]}
      />
      <TextInput
        label="DAYS"
        onChangeText={[Function]}
      />
      <TextInput
        label="HOUR"
        onChangeText={[Function]}
      />
    </View>
    <Button
      onPress={[Function]}
      text="Offer Ride"
    />
  </View>
</View>
`;
